extensions/arc/src/extension.ts

```diff
@@ -57,15 +58,23 @@ export async function activate(context: vscode.ExtensionContext): Promise<arc.IE
 	await checkArcDeploymentExtension();
 
 	return {
-		getRegisteredDataControllers: async () => {
-			return (await treeDataProvider.getChildren())
-				.filter(node => node instanceof ControllerTreeNode)
-				.map(node => (node as ControllerTreeNode).model.info);
-
+		getRegisteredDataControllers: async () => (await treeDataProvider.getChildren())
```

Comment:
This really feels like it should be implemented at a class level rather than an anonymous set of functions as a return statement.

Reply:
for the extension i am working on, even though there is only one method at this time, I still moved it to a separate class. cleaner and easier to test 😆

Reply:
I agree but I will do this later in a different PR because there are a bunch of these in the azdata extension also. I also want to know Charles' views on this

Reply:
Sure, seems reasonable to me. But yeah we can do that in a separate PR I think.

Reply:
Filed #12283 to track this future work.